src/hooks/useFileUpload.tsx

```diff
@@ -342,7 +298,59 @@ export default function useFileUpload(
         });
 
         if (data) {
-          await uploadfile(data, file, associating_id);
+          let lastFilePercent = 0;
```

Comment:
why not have this code inside uploadFile, let's move this back, pass in whatever you need.